rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/DefaultRunConfigurationGenerator.kt

```diff
@@ -5,21 +5,21 @@ import com.intellij.execution.configurations.ConfigurationTypeUtil
 import com.intellij.execution.configurations.UnknownConfigurationType
 import com.intellij.openapi.project.Project
 import com.jetbrains.rdclient.util.idea.LifetimedProjectComponent
-import com.jetbrains.rider.UnityProjectDiscoverer
 import com.jetbrains.rider.plugins.unity.run.configurations.UnityAttachToEditorAndPlayFactory
 import com.jetbrains.rider.plugins.unity.run.configurations.UnityAttachToEditorFactory
 import com.jetbrains.rider.plugins.unity.run.configurations.UnityDebugConfigurationType
 import com.jetbrains.rd.util.reactive.whenTrue
+import com.jetbrains.rider.plugins.unity.UnityHost
 
-class DefaultRunConfigurationGenerator(project: Project, runManager: RunManager, unityProjectDiscoverer: UnityProjectDiscoverer) : LifetimedProjectComponent(project) {
+class DefaultRunConfigurationGenerator(project: Project, runManager: RunManager) : LifetimedProjectComponent(project) {
 
     companion object {
         const val ATTACH_CONFIGURATION_NAME = "Attach to Unity Editor"
         const val ATTACH_AND_PLAY_CONFIGURATION_NAME = "Attach to Unity Editor & Play"
     }
 
     init {
-        unityProjectDiscoverer.hasUnityReference.whenTrue(componentLifetime) {
+        UnityHost.getInstance(project).model.hasUnityReference.whenTrue(componentLifetime) {
```

Comment:
Use project.solution.rdUnityModel directly and also use ProtocolSubscribedProjectComponent as a parent

Reply:
If you worried about polluting the code with runManager getting. define
`val runManager get() = RunManager.getInstance(project)` it's super cheap to get it